src/integrity.rs

```diff
@@ -138,6 +138,23 @@ impl Integrity {
             .result()
     }
 
+    /// Converts a hex string obtained from `to_hex()` to an `Integrity` with a `Hash` containing algorithm and decoded hex string.
+    ///
+    /// # Example
+    ///```
+    /// use ssri::{Integrity, Algorithm};
+    ///
+    /// let expected = Integrity::from(b"hello");
+    /// let hex = String::from("2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824");
+    /// assert_eq!(Integrity::from_hex(hex, Algorithm::Sha256), expected);
+    ///```
+    pub fn from_hex<B: AsRef<[u8]>>(hex: B, algorithm: Algorithm) -> Integrity {
+        let digest = base64::prelude::BASE64_STANDARD.encode(hex::decode(hex).unwrap());
```

Comment:
Unlike `to_hex`, I think this should not `.unwrap()`. You should return a `Result` here. You'll have to add a variant to https://github.com/zkat/ssri-rs/blob/latest/src/errors.rs that wraps [`hex::FromHexError`](https://docs.rs/hex/latest/hex/enum.FromHexError.html)